io/net/bonding.py

```diff
@@ -165,20 +165,47 @@ def bond_fail(self, arg1):
         bond fail
         '''
         for interface in self.host_interfaces:
-            self.log.info("Failing interface %s for mode %s"
-                          % (interface, arg1))
+            if len(self.host_interfaces) > 1:
+                self.log.info("Failing interface %s for mode %s"
+                              % (interface, arg1))
+                cmd = "ifconfig %s down" % interface
+                if process.system(cmd, shell=True, ignore_status=True) != 0:
+                    self.fail("bonding not working when trying to down the\
+                               interface %s " % interface)
+                time.sleep(10)
+                self.ping_check(arg1)
+                cmd = "cat /proc/net/bonding/%s" % self.bond_name
+                process.system_output(cmd, shell=True, verbose=True)
+                cmd = "ifconfig %s up" % interface
+                time.sleep(10)
+                if process.system(cmd, shell=True, ignore_status=True) != 0:
+                    self.fail("bonding not working when trying to up the\
+                              interface %s" % interface)
+                time.sleep(5)
+            else:
+                self.fail("Need a min of 2 host interfaces to to test\
```

Comment:
Theoretically bonding will work even if it has one slave only.
So, we should save the failed case, and fail in the last.